lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -162,9 +164,55 @@ define((require, exports, module) ->
     ch = ContextHandler = {
       contextPreviewExists: false
       UPDATE_DELAY: 1000
+      LABEL_SEQUENCE: [
+        {type: "keyword", value: /^\\label$/}
+        {type: "lparen", value: /^\{$/}
+        {type: "variable.parameter", value: /.*/}
```

Comment:
Dunno anything about equations and environments inside labels, but simple macros definitely do work:
```
\documentclass{article}

\usepackage{amsthm}
\usepackage{hyperref}
\usepackage{cleveref}
  
\newtheorem{thm}[equation]{Theorem}
\newcommand{\foo}[1]{#1#1}
\begin{document}
\begin{thm}
\label{\foo{a}\foo{b}}
This is my theorem\'a
\end{thm}
As proven in theorem \ref{aabb}...
\end{document}
```